cpp/src/arrow/testing/fixed_width_test_util.cc

```diff
@@ -137,7 +137,7 @@ Status NestedListGenerator::AppendNestedList(ArrayBuilder* nested_builder,
     if (type->id() == Type::FIXED_SIZE_LIST) {
       auto* fsl_builder = checked_cast<FixedSizeListBuilder*>(builder);
       assert(list_size == checked_cast<FixedSizeListType&>(*type).list_size());
-      RETURN_NOT_OK(fsl_builder->Append());
+      fsl_builder->UnsafeAppend();
```

Comment:
I don't think we should change this one. This is a testing utility, so we value robustness over performance.